### PR TITLE
Add new command "kubectl set selector"

### PR DESCRIPTION
**Pull request by 0xmichalis**
Continuation of https://github.com/kubernetes/kubernetes/pull/28949

### CODE REVIEW

**Comment by kargakis** [comment]
@kubernetes/kubectl

**Comment by kargakis** [comment]
@k8s-bot gce etcd3 e2e test this

File: pkg/kubectl/cmd/util/factory.go [hunk 1]
```diff
@@ -147,9 +147,9 @@ type ClientAccessFactory interface {
 	// Returns a Printer for formatting objects of the given type or an error.
 	Printer(mapping *meta.RESTMapping, options kubectl.PrintOptions) (kubectl.ResourcePrinter, error)
 	// Pauser marks the object in the info as paused ie. it will not be reconciled by its controller.
-	Pauser(info *resource.Info) (bool, error)
+	Pauser(info *resource.Info) ([]byte, error)
```
Comment (smarterclayton) [inline pkg/kubectl/cmd/util/factory.go]: Describe `[]byte` in the Godoc here.

**Review by smarterclayton — COMMENTED** [review]
One comment, looks good to me.  Might be good for someone else to cast a critical eye.

File: pkg/kubectl/cmd/set/helper.go [hunk 2]
```diff
@@ -117,46 +117,41 @@ type Patch struct {
...
+
+	patch.Patch, patch.Err = strategicpatch.CreateTwoWayMergePatch(patch.Before, patch.After, versioned)
+	return true
+}
+
 // CalculatePatches calls the mutation function on each provided info object, and generates a strategic merge patch for
 // the changes in the object. Encoder must be able to encode the info into the appropriate destination type. If mutateFn
 // returns false, the object is not included in the final list of patches.
-func CalculatePatches(infos []*resource.Info, encoder runtime.Encoder, mutateFn func(*resource.Info) (bool, error)) []*Patch {
+func CalculatePatches(infos []*resource.Info, encoder runtime.Encoder, mutateFn func(*resource.Info) ([]byte, error)) []*Patch {
```
Comment (deads2k) [inline pkg/kubectl/cmd/set/helper.go]: doc is now wrong.  update with what the `mutateFn` returns now.  Also, promote it to a real type and doc it there.

File: pkg/kubectl/cmd/set/set_selector.go [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/runtime"
+)
+
+// SelectorOptions is the start of the data required to perform the operation.  As new fields are added, add them here instead of
+// referencing the cmd.Flags()
+type SelectorOptions struct {
+	fnOptions resource.FilenameOptions
```
Comment (deads2k) [inline pkg/kubectl/cmd/set/set_selector.go]: `fileOptions`.  `fn` looks too much line function.

File: pkg/kubectl/cmd/set/set_selector.go [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+	}
+	cmdutil.AddPrinterFlags(cmd)
+	cmd.Flags().Bool("all", false, "Select all resources in the namespace of the specified resource types")
+	cmd.Flags().Bool("local", false, "If true, set selector will NOT contact api-server but run locally.")
+	cmd.Flags().String("resource-version", "", "If non-empty, the selectors update will only succeed if this is the current resource-version for the object. Only valid when specifying a single resource.")
+	usage := "the resource to update the selectors"
+	cmdutil.AddFilenameOptionFlags(cmd, &options.fnOptions, usage)
+	cmdutil.AddDryRunFlag(cmd)
+	cmdutil.AddRecordFlag(cmd)
+	cmdutil.AddInclude3rdPartyFlags(cmd)
```
Comment (deads2k) [inline pkg/kubectl/cmd/set/set_selector.go]: This isn't needed anymore.

**Comment by deads2k** [comment]
minor comments, then lgtm.

**Comment by kargakis** [comment]
Comments addressed, applying lgtm

**Pull request merged**